### ISSUE TITLE
Different input lengths

### ISSUE THREAD

**Issue by TigranGalstyan** [farizrahman4u/seq2seq#84]
I'm trying to train something like character level transliteration translator. But input length varies from 1 to 20. I want to avoid [bucketing/padding](https://www.tensorflow.org/versions/r0.9/tutorials/seq2seq/index.html#bucketing-and-padding) proposed in tensorflow.
Is there a way to have inputs of different lengths? I think that RNN-s are capable of solving that kind of problems...

**Comment by farizrahman4u** [comment]
Sequences in the same batch must have same input length(because they are represented using matrices). Which means you will need either bucketing or padding. Or you could just train one sequence pair at a time, but that would be slow.

**Comment by TigranGalstyan** [comment]
But can different batches have different output lengths? Because if output length is fixed input does not matter that much.

**Comment by farizrahman4u** [comment]
When traini

**Comment by farizrahman4u** [comment]
When training, make sure that all your sequences end with a stop symbol (like period(.)). Then when you get your output sequences, you could just ignore everything past this symbol. Got it?

**Comment by TigranGalstyan** [comment]
But how can I tell keras to compute loss on only some prefixes of output? Before seeing '<END>' sign :/

**Comment by farizrahman4u** [comment]
Your model should learn to fill the remaining length with the stop symbols. So it should be the same way in your training data.

**Comment by TigranGalstyan** [comment]
But the problem is, that my word lengths are from 1 to 20 and majority is less then 10. It's becoming hard for a model to learn something except putting '<END>' symbols everywhere. I was just wondering is there any way to have different output lengths to overcome this problem.

**Comment by farizrahman4u** [comment]
You can write a custom loss function, but its much easier the way I said before.

**Comment by dead** [comment]
Iirc you can make keras compute loss on only some outputs using sample_weight
